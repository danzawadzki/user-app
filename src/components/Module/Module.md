Default module (higher order component):

```jsx
<Module>Test</Module>
```

Module with button in the top right corner.

```jsx
<Module actions={<button>Action</button>}>Test</Module>
```
