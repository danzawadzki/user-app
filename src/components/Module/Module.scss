@import '../../assets/styles/colors.scss';
@import '../../assets/styles/base.scss';
@import '../../assets/styles/minor.scss';

.Module {
	border-radius: $radius--default;
	box-shadow: $shadow--default;
	padding: 2rem;
	font-family: Montserrat, sans-serif;
}
