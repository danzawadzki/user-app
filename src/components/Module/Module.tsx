import * as React from 'react';
import './Module.scss';

export interface IModule {
	/** Module child html node. */
	children: JSX.Element | JSX.Element[] | string;
}

/**
 * Module component - higher order component to display UI elements
 * inside white module with rounded corners and shadow under.
 *
 * @version 1.0.0
 */
const Module = ({ children }: IModule) => <div className="Module">{children}</div>;

export default Module;
