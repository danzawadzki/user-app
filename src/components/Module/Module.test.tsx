import React from 'react';
import { shallow } from 'enzyme';
import Module from './Module';

describe('Module component: ', () => {
	// Shallow rendered component
	const content = <p>Test</p>;

	const component = shallow(<Module>{content}</Module>);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render correctly', () => {
		expect(component.find('.Module')).toBeDefined();
	});

	it('should render with child component', () => {
		expect(component.html()).not.toBe(null);
		expect(component.find(content)).toBeTruthy();
	});
});
