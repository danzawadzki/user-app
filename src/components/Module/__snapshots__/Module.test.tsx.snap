// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module component:  should render match snapshot 1`] = `
<div
  className="Module"
>
  <p>
    Test
  </p>
</div>
`;
